libraries/lib-files/FileNames.cpp

```diff
@@ -224,8 +230,48 @@ wxString FileNames::LowerCaseAppNameInPath( const wxString & dirIn){
    return dir;
 }
 
+FilePath FileNames::ConfigDir()
+{
+#if defined(__WXGTK__)
+   if (gOldUnixDataDir.empty())
+      gOldUnixDataDir = wxFileName::GetHomeDir() + wxT("/.audacity-data");
+#endif
+
+   if (gConfigDir.empty())
+   {
+      wxFileName exePath(PlatformCompatibility::GetExecutablePath());
+#if defined(__WXMAC__)
+      // Path ends for example in "Audacity.app/Contents/MacOSX"
+      // just remove the MacOSX part.
+      exePath.RemoveLastDir();
+#endif
+      wxFileName portablePrefsPath(exePath.GetPath(), wxT("Portable Settings"));
+      if (::wxDirExists(portablePrefsPath.GetFullPath()))
+      {
+         // Use "Portable Settings" folder
+         gConfigDir = portablePrefsPath.GetFullPath();
+#if defined(__WXGTK__)
+      } else if (::wxDirExists(gOldUnixDataDir))
+      {
+         // Use old user data dir folder
+         gConfigDir = gOldUnixDataDir;
+#endif
+      } else {
+         // Use OS-provided user data dir folder
+         wxString configDir(wxStandardPaths::Get().GetUserConfigDir() + wxT("/audacity"));
```

Comment:
I am not sure, if I follow? `GetUserDataDir()` will return `~/.local/share`, not `~/.config`. +1 on using `AUDACITY_NAME` instead of a hardcoded string.

Reply:
Docs are confusing: `Only use this method if you have a single configuration file to put in this directory, otherwise [GetUserDataDir()](https://docs.wxwidgets.org/3.1/classwx_standard_paths.html#a5b9d1b1addc3e4ce30ccb3817cbbe19b) is more appropriate as the latter adds appinfo to the path, unlike this function.` https://docs.wxwidgets.org/3.1/classwx_standard_paths.html#a0561d70f7da648e10b25bf9679309da3

Reply:
I agree, that the docs are confusing. I am currently rewriting the implementation based on your `enum DirTarget` idea.

Actually the `wxwidgets` `GetUserWhateverDir` methods don't map to corresponding `XDG_WHATEVER_HOME` directories, so I decided to just implement the correct XDG Base Directory logic for unix and keep the windows case exactly as it currently is.

More info later. Btw, are we sure that `__WXGTK__` is the correct preprocessor macro to distinguish unix? I just kept it from the original PR, but I am not 100% sure.

Reply:
> are we sure that __WXGTK__ 

We use it for now, at least, so it's good to go!